src/webui/gitutils.jl

```diff
@@ -242,12 +252,17 @@ function gettreesha(
 )
     return try
         url = cloneurl(r)
+        if !is_safe_clone_url(url)
+            throw(ArgumentError("Invalid or unsafe clone URL"))
+        end
         mktempdir() do dir
             dest = joinpath(dir, r.name)
             withpasswd(url) do url, env
-                run(Cmd(`git clone --bare $url $dest`; env))
+                # let Cmd interpolate strings safely
+                run(Cmd(Cmd(String["git", "clone", "--bare", string(url), dest]); env))
```

Comment:
Why are two calls to `Cmd()` needed here?

Reply:
Yeah...